### ISSUE TITLE
[TODO] Server side command configuration

### ISSUE THREAD

**Issue by jarmentor** [gbowne1/TwitchBot#71]
**Is your feature request related to a problem? Please describe.**
Yes. As noted [here](https://github.com/gbowne1/TwitchBot/pull/62#issuecomment-1962423338). Both the client and server would need to be updated each time a command was added. This is suboptimal, to be sure.

**Describe the solution you'd like**
Perhaps a server route hit when the component mounts (not sure what this is called now) or when the app starts? To get an initial config object of available/supported commands and associated API routes?

**Describe alternatives you've considered**
- None. 🚩 — This seems the way to do it though?

**Comment by gbowne1** [comment]
I agree with that solution.

**Comment by gbowne1** [comment]
@jarmentor 

Am going to completely refactor pretty much everything in src/server.  And so I will ust push up new code that actually works with MongoDB and Twitch and Discord. 

A lot of our issues are related to this poor design we've been using.

**Comment by jarmentor** [comment]
Greetings! I haven't had as much time for open source stuff as I'd like of late, but this sounds great! I'm happy to contribute where I can, this sounds good to me though

**Comment by gbowne1** [comment]
Yes, I just really discovered how pretty poorly this whole project is.. so yeah I'll just push up a working server by Monday. Long weekend so, I'll be able to push through it by Monday USA time.

**Comment by gbowne1** [comment]
This is what you'll see now when the server's running on http://localhost:3001

![Image](https://github.com/user-attachments/assets/e1812e1f-d48a-43dc-bbb7-356b9cbaf6e5)

**Comment by gbowne1** [comment]
@jarmentor pushed up the first part of the refactoring.  `node server.js` should work fine.  I need to work on nodemon.  It connects to the database twitchbot on localhost fine.  Server starts fine. Twitch would need someone to make a test account or figure out how to test the bot with the old justin.tv api I didn't get a chance to do it because you have to register a bot with twitch.

**Comment by gbowne1** [comment]
commands are now dynamic. You can add commands via adding it to the commands directory jusr make sure its in es module syntax
